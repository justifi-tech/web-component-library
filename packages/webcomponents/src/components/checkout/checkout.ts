export interface ICheckout {
  id: string;
  type: string;
  page_info: null;
  data: {
    payment_intent_id: string;
    account_id: string;
    platform_account_id: string;
    payment_amount: string;
    payment_client_id: string;
    payment_description: string;
    payment_methods: {
      id: string;
      status: string;
      invalid_reason: null;
      name: string;
      brand: string;
      acct_last_four: string;
      month: string;
      year: string;
      address_line1_check: string;
      address_postal_code_check: string;
      bin_details: null;
    }[];
    payment_method_group_id: string;
    payment_settings: {
      ach_payments: boolean;
      bnpl_payments: boolean;
      credit_card_payments: boolean;
    };
    bnpl?: {
      provider: string;
      provider_client_id: string;
      provider_mode: string;
      provider_checkout_url: string;
      provider_order_id: string;
      provider_api_version: string;
    };
  };
}

export class Checkout implements ICheckout {
  id: string;
  type: string;
  page_info: null;
  data: {
    payment_intent_id: string;
    account_id: string;
    platform_account_id: string;
    payment_amount: string;
    payment_client_id: string;
    payment_description: string;
    payment_methods: {
      id: string;
      status: string;
      invalid_reason: null;
      name: string;
      brand: string;
      acct_last_four: string;
      month: string;
      year: string;
      address_line1_check: string;
      address_postal_code_check: string;
      bin_details: null;
    }[];
    payment_method_group_id: string;
    payment_settings: {
      ach_payments: boolean;
      bnpl_payments: boolean;
      credit_card_payments: boolean;
    };
    bnpl?: {
      provider: string;
      provider_client_id: string;
      provider_mode: string;
      provider_checkout_url: string;
      provider_order_id: string;
      provider_api_version: string;
    };
  };

  constructor(data: ICheckout) {
    Object.assign(this, data);
  }
}
