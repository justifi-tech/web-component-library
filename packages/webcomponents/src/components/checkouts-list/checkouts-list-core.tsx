import { Component, h, Prop, State, Watch, Event, EventEmitter } from '@stencil/core';
import { Checkout, ICheckoutsParams, PagingInfo, SubAccount, pagingDefaults } from '../../api';
import { ComponentError } from '../../api/ComponentError';
import { tableExportedParts } from '../../ui-components/table/exported-parts';
import { onFilterChange } from '../../ui-components/filters/utils';
import { StyledHost, TableEmptyState, TableErrorState, TableLoadingState } from '../../ui-components';
import CheckoutsTable, { defaultColumnsKeys } from './checkouts-table';

@Component({
  tag: 'checkouts-list-core'
})
export class CheckoutsListCore {
  @Prop() getCheckouts: Function;
  @Prop() getSubAccounts: Function;
  @Prop() columns: string = defaultColumnsKeys;

  @State() checkouts: Checkout[] = [];
  @State() checkoutsTable: CheckoutsTable;
  @State() subAccounts: SubAccount[] = [];
  @State() loading: boolean = true;
  @State() errorMessage: string;
  @State() paging: PagingInfo = pagingDefaults;
  @State() params: ICheckoutsParams = {};

  @Watch('params')
  @Watch('getCheckouts')
  @Watch('getSubAccounts')
  @Watch('columns')
  updateOnPropChange() {
    this.fetchCheckouts();
  }

  @Event({
    eventName: 'checkout-row-clicked',
    bubbles: true,
  }) rowClicked: EventEmitter<Checkout>;

  @Event({ eventName: 'error-event' }) errorEvent: EventEmitter<ComponentError>;

  componentWillLoad() {
    if (this.getCheckouts && this.getSubAccounts) {
      this.fetchCheckouts();
    }
  }

  fetchCheckouts(): void {
    this.loading = true;

    this.getCheckouts({
      params: this.params,
      onSuccess: ({ checkouts, pagingInfo }) => {
        this.checkouts = checkouts;
        this.paging = pagingInfo;
        this.checkoutsTable = new CheckoutsTable(this.checkouts, this.columns);
        this.loading = false; // temp
      },
      onError: ({ error, code, severity }) => {
        this.errorMessage = error;
        this.errorEvent.emit({
          errorCode: code,
          message: error,
          severity,
        });
        this.loading = false;
      },
    });
  }

  async fetchSubAccounts(): Promise<void> {
    this.getSubAccounts({
      params: this.subAccountParams,
      onSuccess: ({ subAccounts }) => {
        this.subAccounts = subAccounts;
        this.checkouts = this.checkouts.map((checkout) => {
          checkout.sub_account_name = this.subAccounts.find((subAccount) => subAccount.id === checkout.account_id)?.name;
          return checkout;
        });
        this.loading = false;
      },
      onError: ({ error, code, severity }) => {
        this.errorMessage = error;
        this.errorEvent.emit({
          errorCode: code,
          message: error,
          severity,
        });
        this.loading = false;
      },
    });
  }

  handleClickPrevious = (beforeCursor: string) => {
    const newParams: any = { ...this.params };
    delete newParams.after_cursor;
    this.params = { ...newParams, before_cursor: beforeCursor };
  };

  handleClickNext = (afterCursor: string) => {
    const newParams: any = { ...this.params };
    delete newParams.before_cursor;
    this.params = { ...newParams, after_cursor: afterCursor };
  };

  rowClickHandler = (e) => {
    const clickedCheckoutID = e.target.closest('tr').dataset.rowEntityId;
    if (!clickedCheckoutID) return;
    this.rowClicked.emit(this.checkouts.find((checkout) => checkout.id === clickedCheckoutID));
  };

  setParamsOnChange = (name: string, value: string) => {
    let newParams = { [name]: value };
    this.params = onFilterChange(newParams, this.params);
  }

  clearParams = () => {
    this.errorMessage = '';
    this.params = {};
  }

  get subAccountParams() {
    let accountIdNumbers = this.checkouts.map((checkout) => checkout.account_id);
    let uniqueAccountIds = [...new Set(accountIdNumbers)];
    let accountIdString = uniqueAccountIds.join(',');
    return { sub_account_id: accountIdString };
  }

  get entityId() {
    return this.checkouts.map((checkout) => checkout.id);
  }

  get showEmptyState() {
    return !this.loading && !this.errorMessage && this.checkoutsTable.rowData.length < 1;
  }

  get showErrorState() {
    return !this.loading && !!this.errorMessage;
  }

  get showRowData() {
    return !this.showEmptyState && !this.showErrorState && !this.loading;
  }

  render() {
    return (
      <StyledHost exportparts={tableExportedParts}>
        <checkouts-list-filters
          params={this.params}
          setParamsOnChange={this.setParamsOnChange}
          clearParams={this.clearParams}
        />
        <div class="table-wrapper">
          <table class="table table-hover">
            <thead class="table-head sticky-top" part="table-head">
              <tr class="table-light text-nowrap" part="table-head-row">
                {this.checkoutsTable?.columnData?.map((column) => (
                  <th part="table-head-cell" scope="col" title={column.title}>
                    {column.label}
                  </th>
                ))}
              </tr>
            </thead>
            <tbody class="table-body" part="table-body">
              {/* fix this, not working with new checkoutsTable */}
              <TableLoadingState
                columnSpan={this.checkoutsTable?.columnData.length}
                isLoading={this.loading}
              />
              <TableEmptyState
                isEmpty={this.showEmptyState}
                columnSpan={this.checkoutsTable?.columnData.length}
              />
              <TableErrorState
                columnSpan={this.checkoutsTable?.columnData.length}
                errorMessage={this.errorMessage}
              />
              {this.showRowData && this.checkoutsTable?.rowData.map((data) => (<tr>{data}</tr>))}
            </tbody>
            {this.paging && (
              <tfoot class="sticky-bottom">
                <tr class="table-light align-middle">
                  <td part="pagination-bar" colSpan={this.checkoutsTable?.columnData?.length}>
                    <pagination-menu
                      paging={{
                        ...this.paging,
                        handleClickPrevious: this.handleClickPrevious,
                        handleClickNext: this.handleClickNext,
                      }}
                      params={this.params}
                    />
                  </td>
                </tr>
              </tfoot>
            )}
          </table>
        </div>
      </StyledHost>
    );
  }
}
