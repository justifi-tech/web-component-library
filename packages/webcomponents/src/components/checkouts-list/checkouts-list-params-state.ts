import { createStore } from '@stencil/store';
import { CheckoutsQueryParams } from '../../api';

const filterParamsInitialState = {};
const propsParamsInitialState = {};

const filterParamsStore = createStore<CheckoutsQueryParams>(() => filterParamsInitialState);
const { state: filterParams, on: onQueryParamsChange } = filterParamsStore;

const propsParamsStore = createStore<CheckoutsQueryParams>(() => propsParamsInitialState);
const { state: propsParams } = propsParamsStore;

const getRequestParams = (): CheckoutsQueryParams => {
  return {
    ...filterParams,
    ...propsParams
  };
};

const clearParams = () => {
  filterParamsStore.reset();
}

export {
  filterParams,
  propsParams,
  onQueryParamsChange,
  clearParams,
  getRequestParams
};
