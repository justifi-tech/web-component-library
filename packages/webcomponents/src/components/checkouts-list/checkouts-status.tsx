
export const MapCheckoutStatusToBadge = (status: string) => {
  switch (status) {
    case 'created':
      return "<span class='badge bg-primary' title='The checkout has been created'>Created</span>";
    case 'completed':
      return "<span class='badge bg-success' title='The checkout has been completed'>Completed</span>";
    case 'attempted':
      return "<span class='badge bg-secondary' title='The checkout has been attempted'>Attempted</span>";
    case 'expired':
      return "<span class='badge bg-danger' title='The checkout has expired'>Expired</span>";
  };
};
