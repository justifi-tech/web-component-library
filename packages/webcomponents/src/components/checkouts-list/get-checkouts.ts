import { Checkout } from '../../api';
import { ComponentErrorSeverity } from '../../api/ComponentError';
import { getErrorCode, getErrorMessage } from '../../api/services/utils';

export const makeGetCheckoutsList =
  ({ authToken, service, apiOrigin }) =>
    async ({ params, onSuccess, onError }) => {
      try {
        const response = await service.fetchCheckoutsList(authToken, params, apiOrigin);

        if (!response.error) {
          const pagingInfo = {
            ...response.page_info,
          };

          const checkouts =
            response.data?.map((dataItem) => new Checkout(dataItem)) || [];

          onSuccess({ checkouts, pagingInfo });
        } else {
          const responseError = getErrorMessage(response.error);
          const code = getErrorCode(response.error?.code);
          return onError({
            error: responseError,
            code,
            severity: ComponentErrorSeverity.ERROR,
          });
        }
      } catch (error) {
        const code = getErrorCode(error?.code);
        return onError({
          error: error.message || error,
          code,
          severity: ComponentErrorSeverity.ERROR,
        });
      }
    };
