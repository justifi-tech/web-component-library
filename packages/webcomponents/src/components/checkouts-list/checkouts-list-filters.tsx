import { Component, h, Prop } from '@stencil/core';
import { ICheckoutStatus } from '../../api';
import { filterParams, propsParams, clearParams } from './checkouts-list-params-state';
import { StyledHost } from '../../ui-components';
import { checkoutsListFilterMenu, paymentModeCheckoutsListFilterParam, statusCheckoutsListFilterParam } from '../../styles/parts';

@Component({
  tag: 'justifi-checkouts-list-filters',
  shadow: true
})
export class CheckoutsListFilters {
  @Prop() status?: ICheckoutStatus;
  @Prop() paymentMode?: string;

  componentWillLoad() {
    const propsToSet = {
      status: this.status,
      payment_mode: this.paymentMode,
    };

    Object.entries(propsToSet).forEach(([key, value]) => {
      if (value) {
        propsParams[key] = value;
      }
    });
  }
  
  setParamsOnChange = (name: string, value: string) => {
    filterParams[name] = value;
  }

  get checkoutStatusOptions(): { label: string, value: ICheckoutStatus | '' }[] {
    return [
      { label: 'All', value: '' },
      { label: 'Created', value: 'created' },
      { label: 'Completed', value: 'completed' },
      { label: 'Attempted', value: 'attempted' },
      { label: 'Expired', value: 'expired' },
    ]
  }

  get checkoutPaymentModeOptions(): { label: string, value: string }[] {
    return [
      { label: 'All', value: '' },
      { label: 'E-commerce', value: 'ecom' },
      { label: 'BNPL', value: 'bnpl' },
    ]
  }

  render() {
    const filterMenuParams = { ...filterParams }

    return (
      <StyledHost>
        <table-filters-menu params={filterMenuParams} clearParams={clearParams} part={checkoutsListFilterMenu}>
          <div class="grid-cols-2 gap-3 p-1">
            <div class="p-2">
              <form-control-select
                name="status"
                label="Status"
                options={this.checkoutStatusOptions}
                inputHandler={this.setParamsOnChange}
                defaultValue={this.status || filterParams.status}
                disabled={!!this.status}
                part={statusCheckoutsListFilterParam}
              />
            </div>
            <div class="p-2">
              <form-control-select
                name="payment_mode"
                label="Payment Mode"
                options={this.checkoutPaymentModeOptions}
                inputHandler={this.setParamsOnChange}
                defaultValue={this.paymentMode || filterParams.payment_mode}
                disabled={!!this.paymentMode}
                part={paymentModeCheckoutsListFilterParam}
              />
            </div>
          </div>
        </table-filters-menu>
      </StyledHost>
    )
  }
}
