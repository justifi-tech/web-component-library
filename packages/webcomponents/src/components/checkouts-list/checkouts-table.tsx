import { h } from '@stencil/core';
import { formatDate, formatTime } from '../../utils/utils';
import { MapCheckoutStatusToBadge } from './checkouts-status';
import { getAlternateTableCellPart, tableHeadCell } from '../../styles/parts';
import { Checkout } from '../../api/Checkout';

export const defaultColumnsKeys = 'created_at,payment_amount,payment_description,payment_mode,status';

export const checkoutTableColumns = {
  created_at: () => (
    <th part={tableHeadCell} scope="col" title="The date each checkout occurred">
      Processed On
    </th>
  ),
  payment_amount: () => (
    <th part={tableHeadCell} scope="col" title="The dollar amount of each checkout">
      Payment Amount
    </th>
  ),
  payment_description: () => (
    <th part={tableHeadCell} scope="col" title="The description of each checkout">
      Payment Description
    </th>
  ),
  sub_account_name: () => (
    <th part={tableHeadCell} scope="col" title="The sub account associated with the checkout">
      Sub Account
    </th>
  ),
  payment_mode: () => (
    <th part={tableHeadCell} scope="col" title="The payment mode of each checkout">
      Payment Mode
    </th>
  ),
  status: () => (
    <th part={tableHeadCell} scope="col" title="The current status of each checkout">
      Status
    </th>
  ),
}

export const checkoutTableCells = {
  created_at: (checkout: Checkout, index: number) => (
    <td part={getAlternateTableCellPart(index)}>
      <div class="fw-bold">{formatDate(checkout.created_at)}</div>
      <div class="fw-bold">{formatTime(checkout.created_at)}</div>
    </td>
  ),
  payment_amount: (checkout: Checkout, index: number) => (
    <td part={getAlternateTableCellPart(index)}>{checkout.formatPaymentAmount(checkout.payment_amount, true)}</td>
  ),
  payment_description: (checkout: Checkout, index: number) => (
    <td part={getAlternateTableCellPart(index)}>{checkout.payment_description}</td>
  ),
  sub_account_name: (checkout: Checkout, index: number) => (
    <td part={getAlternateTableCellPart(index)}>{checkout.sub_account_name}</td>
  ),
  payment_mode: (checkout: Checkout, index: number) => (
    <td part={getAlternateTableCellPart(index)}>{checkout.payment_mode}</td>
  ),
  status: (checkout: Checkout, index: number) => (
    <td part={getAlternateTableCellPart(index)} innerHTML={MapCheckoutStatusToBadge(checkout.status)}></td>
  ),
};
