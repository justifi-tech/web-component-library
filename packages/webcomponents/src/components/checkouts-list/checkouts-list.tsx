import { Component, Event, EventEmitter, h, Prop, State, Watch } from '@stencil/core';
import { makeGetCheckouts } from '../../actions/checkout/get-checkouts';
import { checkPkgVersion } from '../../utils/check-pkg-version';
import JustifiAnalytics from '../../api/Analytics';
import { CheckoutService } from '../../api/services/checkout.service';
import { ComponentErrorCodes, ComponentErrorSeverity } from '../../api/ComponentError';
import { SubAccountService } from '../../api/services/subaccounts.service';
import { makeGetSubAccounts } from '../../actions/sub-account/get-subaccounts';
import { StyledHost } from '../../ui-components';
import { defaultColumnsKeys } from './checkouts-table';
import { ComponentErrorEvent } from '../../api/ComponentEvents';

@Component({
  tag: 'justifi-checkouts-list',
  shadow: true
})

export class CheckoutsList {
  @State() getCheckouts: Function;
  @State() getSubAccounts: Function;
  @State() errorMessage: string = null;

  @Prop() accountId: string;
  @Prop() authToken: string;
  @Prop() columns?: string = defaultColumnsKeys;

  @Event({ eventName: 'error-event' }) errorEvent: EventEmitter<ComponentErrorEvent>;

  analytics: JustifiAnalytics;

  componentWillLoad() {
    checkPkgVersion();
    this.analytics = new JustifiAnalytics(this);
    this.initializeGetData();
  }

  disconnectedCallback() {
    this.analytics.cleanup();
  }

  @Watch('accountId')
  @Watch('authToken')
  propChanged() {
    this.initializeGetData();
  }

  private initializeGetData() {
    this.initializeGetCheckouts();
    this.initializeGetSubAccounts();
  }

  private initializeGetCheckouts() {
    if (this.accountId && this.authToken) {
      this.getCheckouts = makeGetCheckouts({
        accountId: this.accountId,
        authToken: this.authToken,
        service: new CheckoutService()
      });
    } else {
      this.errorMessage = 'Account ID and Auth Token are required';
      this.errorEvent.emit({
        errorCode: ComponentErrorCodes.MISSING_PROPS,
        message: this.errorMessage,
        severity: ComponentErrorSeverity.ERROR,
      });
    }
  }

  private initializeGetSubAccounts() {
    if (this.accountId && this.authToken) {
      this.getSubAccounts = makeGetSubAccounts({
        accountId: this.accountId,
        authToken: this.authToken,
        service: new SubAccountService()
      });
    }
  }

  handleErrorEvent = (event) => {
    this.errorMessage = event.detail.message;
    this.errorEvent.emit(event.detail);
  };

  render() {
    return (
      <StyledHost>
        <checkouts-list-core
          getCheckouts={this.getCheckouts}
          getSubAccounts={this.getSubAccounts}
          onError-event={this.handleErrorEvent}
          columns={this.columns}
        />
      </StyledHost>
    );
  }
}
