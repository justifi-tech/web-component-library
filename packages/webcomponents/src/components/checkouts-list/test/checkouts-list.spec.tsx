import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { CheckoutsList } from '../checkouts-list';
import { CheckoutsListCore } from '../checkouts-list-core';
import { CheckoutService } from '../../../api/services/checkout.service';
import JustifiAnalytics from '../../../api/Analytics';
jest.mock('../../../api/services/checkout.service');

beforeEach(() => {
  // Bypass Analytics to avoid errors. Analytics attaches events listeners to HTML elements
  // which are not available in Jest/node environment
  // @ts-ignore
  JustifiAnalytics.prototype.trackCustomEvents = jest.fn();
});

describe('checkouts-list', () => {
  it('renders an error message when accountId and authToken are not provided', async () => {
    const page = await newSpecPage({
      components: [CheckoutsList, CheckoutsListCore],
      template: () => <justifi-checkouts-list />,
    });
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });

  it('renders an error message when accountId is not provided', async () => {
    const page = await newSpecPage({
      components: [CheckoutsList, CheckoutsListCore],
      template: () => <justifi-checkouts-list auth-token="abc" />,
    });
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });

  it('renders an error message when authToken is not provided', async () => {
    const page = await newSpecPage({
      components: [CheckoutsList, CheckoutsListCore],
      template: () => <justifi-checkouts-list account-id="abc" />,
    });
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });

  it('emit an error event when accountId and authToken are not provided', async () => {
    const errorEvent = jest.fn();
    const page = await newSpecPage({
      components: [CheckoutsList, CheckoutsListCore],
      template: () => <justifi-checkouts-list onError-event={errorEvent} />,
    });
    await page.waitForChanges();
    expect(errorEvent).toHaveBeenCalled();
  });

  it('emit an error event when fetch fails', async () => {
    CheckoutService.prototype.fetchCheckouts = jest.fn().mockRejectedValue(new Error('Fetch error'));

    const errorEvent = jest.fn();
    const page = await newSpecPage({
      components: [CheckoutsList, CheckoutsListCore],
      template: () => (
        <justifi-checkouts-list
          account-id="abc"
          auth-token="abc"
          onError-event={errorEvent}
        />
      ),
    });
    await page.waitForChanges();
    expect(errorEvent).toHaveBeenCalledWith(
      expect.objectContaining({
        detail: {
          errorCode: 'fetch-error',
          message: 'Fetch error',
          severity: 'error',
        }
      })
    );
  });
});
