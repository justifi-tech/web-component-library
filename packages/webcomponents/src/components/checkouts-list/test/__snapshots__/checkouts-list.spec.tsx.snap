// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkouts-list renders an error message when accountId and authToken are not provided 1`] = `
<justifi-checkouts-list>
  <template shadowrootmode="open">
    <main class="d-flex justify-content-center p-4 text-center" part="detail-empty-state" style="font-size: 1.2rem;">
      <div>
        Account ID and Auth Token are required
      </div>
    </main>
  </template>
</justifi-checkouts-list>
`;

exports[`checkouts-list renders an error message when accountId is not provided 1`] = `
<justifi-checkouts-list auth-token="abc">
  <template shadowrootmode="open">
    <main class="d-flex justify-content-center p-4 text-center" part="detail-empty-state" style="font-size: 1.2rem;">
      <div>
        Account ID and Auth Token are required
      </div>
    </main>
  </template>
</justifi-checkouts-list>
`;

exports[`checkouts-list renders an error message when authToken is not provided 1`] = `
<justifi-checkouts-list account-id="abc">
  <template shadowrootmode="open">
    <main class="d-flex justify-content-center p-4 text-center" part="detail-empty-state" style="font-size: 1.2rem;">
      <div>
        Account ID and Auth Token are required
      </div>
    </main>
  </template>
</justifi-checkouts-list>
`;
